tests.py

```diff
@@ -64,6 +64,54 @@ def test_actual_references_not_copies(self):
         assert result["c"].__subject__ is result["a"]
         assert result["d"].__subject__ is result["a"]
 
+    def test_ignored_sibling_attributes(self):
+        json = {
+            "a": {"type": "object", "properties": {"foo": {"type": "string"}}},
+            "b": {"extra": "foobar", "$ref": "#/a"},
+            "c": {"extra": { "more": "bar", "$ref": "#/a"} },
+            "d": {"deep":  { "$ref": "#/a", "more": { "$ref": "#/b" } } },
+        }
+        result = JsonRef.replace_refs(json)
+        # because we deepcopy, use '==' instead of 'is"
+        assert result["b"] == {
+            "extra": "foobar",
+            "type": "object",
+            "properties": {"foo": {"type": "string"}},
+        }
+        assert result["c"] == {
+            "extra": {
+                "more": "bar",
+                "type": "object",
+                "properties": {"foo": {"type": "string"}}
+            }
+        }
+        assert result["d"] == {
+            "deep": {
+                "more": {
+                    "extra": "foobar",
+                    "type": "object",
+                    "properties": {"foo": {"type": "string"}},
+                },
+                "type": "object",
+                "properties": {"foo": {"type": "string"}}
+            }
+        }
+
+    def test_extra_sibling_attributes_list_ref(self):
+        json = {
+            "a": ["target"],
+            "b": {"extra": "foobar", "$ref": "#/a"},
+        }
+        result = JsonRef.replace_refs(json)
+        with pytest.raises(JsonRefError) as excinfo:
```

Comment:
I think this is a bad behavior, even if the extra attributes are opt-in. This clearly violates the spec stipulation that extra attributes should be ignored. I think when turning on a merge-attributes mode, it would only do anything if the ref points to a dictionary, and just ignore them otherwise.

Reply:
fair enough